lib/deploy/upload.js

```diff
@@ -24,12 +24,21 @@ function upload(exp, data) {
         },
         formData: data
       }, (err, response, body) => {
-        if (err) return reject(err);
-
-        resolve(JSON.parse(body));
+        if (err) {
+          reject({ code: 'network.error', message: err.toString() });
+        } else if ( response.statusCode >= 400 ) {
+          if (response.headers['content-type'] === 'application/json') {
+            reject(JSON.parse(body));
+          } else {
+            reject({ code: 'unknown.error', message: body.toString() });
+          }
+        } else {
+          resolve(JSON.parse(body));
+        }
       });
   })
-  .then(res => exp.getContent(res.uuid));
+  .then(res => exp.getContent(res.uuid))
+  .catch( err => err );
```

Comment:
@jrd261 the various upload methods below all call this method but the `.catch` treats errors differently with different messages, so "yes", the consumer of this `throw`s the error and logs a message to the console that's different depending on context.